crates/wasi-runtime-config/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::Result;
+use std::collections::HashMap;
+
+mod gen_ {
+    wasmtime::component::bindgen!({
+        path: "wit",
+        world: "wasi:config/imports",
+        trappable_imports: true,
+    });
+}
+use self::gen_::wasi::config::runtime as generated;
+
+/// Capture the state necessary for use in the `wasi-runtime-config` API implementation.
+pub struct WasiRuntimeConfigCtx {
+    runtime_config: HashMap<String, String>,
+}
+
+impl WasiRuntimeConfigCtx {
+    /// Create a new context.
+    pub fn new<S, I>(config: I) -> Self
+    where
+        S: Into<String>,
+        I: IntoIterator<Item = (S, S)>,
+    {
+        Self {
+            runtime_config: config
+                .into_iter()
+                .map(|(k, v)| (k.into(), v.into()))
+                .collect(),
+        }
+    }
```

Comment:
I think it would be more ergonomic and idiomatic to turn this function into a [`FromIterator`](https://doc.rust-lang.org/nightly/std/iter/trait.FromIterator.html) trait implementation, derive the `Default` trait for this type, make `new` return the default, and also add a `fn define(&mut self, key: impl Into<String>, value: impl Into<String>) -> &mut Self` builder-style method.